libs/shared/src/services/core/analytics.service.ts

```diff
@@ -35,4 +35,20 @@ export class AnalyticsService {
       }
     });
   }
+
+  // Method to track when users play a video
+  public trackVideoPlay(videoId: string) {
+    this.firebaseAnalytics.logEvent({
+      name: 'video_play',
+      params: { video_id: videoId, app_version: APP_VERSION },
```

Comment:
nit(non-blocking): `app_version` will automatically be tracked in various other places so no need to explictily send with each analytics request, but very nice to have the video_id getting passed.